specification/security/resource-manager/Microsoft.Security/preview/2023-07-01-preview/integrations.json

```diff
@@ -0,0 +1,650 @@
+			"type": "string",
+			"description": "Defines whether the integration is management scope integration (master connector as a single scope or management scope)"
+		  },
+		  "metadata": {
+			"$ref": "#/definitions/IntegrationMetadata"
+		  },
+		  "integrationAdditionalData": {
+			"type": "object",
+			"description": "Additional data",
+			"$ref": "#/definitions/IntegrationAdditionalData"
+		  }
+		},
+		"required": [
+		  "displayName",
+		  "integrationType"
+		]
+	  },
+	  "IntegrationRootDetails": {
+		"type": "object",
+		"description": "The integration root details",
+		"properties": {
+		  "resourceId": {
+			"description": "The integration root resource Id",
+			"type": "string",
+			"readOnly": true
+		  },
+		  "distance": {
+			"description": "The number of nodes between integration and the root integration resource",
+			"type": "number",
+			"readOnly": true
+		  }
+		}
+	  },
+	  "IntegrationMetadata": {
+		"type": "object",
+		"description": "The integration metadata",
+		"properties": {
+		  "createdBy": {
+			"description": "Integration Created by object id (GUID)",
+			"type": "string",
+			"readOnly": true
+		  },
+		  "createdOn": {
+			"description": "Integration creation date",
+			"type": "string",
+			"format": "date-time",
+			"readOnly": true
+		  },
+		  "updatedBy": {
+			"description": "Integration last updated by object id (GUID)",
+			"type": "string",
+			"readOnly": true
+		  },
+		  "updatedOn": {
+			"description": "Integration last update date",
+			"type": "string",
+			"format": "date-time",
+			"readOnly": true
+		  }
+		}
+	  },
+	  "IntegrationAdditionalData": {
+		"type": "object",
+		"description": "The integration additional data (abstract)",
+		"discriminator": "integrationPlatform",
+		"properties": {
+		  "integrationPlatform": {
+			"type": "string",
+			"description": "Describes the integration platform"
+		  }
+		},
+		"required": [
+		  "integrationPlatform"
+		]
+	  },
+	  "ServiceNowAdditionalData": {
+		"description": "Service Now specific additional data",
+		"allOf": [
+		  {
+			"$ref": "#/definitions/IntegrationAdditionalData"
+		  },
+		  {
+			"properties": {
+			  "instanceUrl": {
+				"description": "Integration creation date",
+				"type": "string"
+			  },
+			  "clientId": {
+				"description": "Integration creation date",
+				"type": "string"
+			  },
+			  "clientSecret": {
+				"description": "Integration creation date",
+				"type": "string"
+			  },
+			  "tableMetadata": {
+				"type": "array",
+				"items": {
+				  "$ref": "#/definitions/ServiceNowTableMetadata"
+				}
+			  }
+			}
+		  }
+		],
+		"x-ms-discriminator-value": "ServiceNow"
+	  },
+	  "ServiceNowAuthenticationDetails": {
+		"description": "Service Now authentication details",
+		"type": "object",
+		"properties": {
+		  "userName": {
+			"type": "string",
+			"description": "The integration user name"
+		  },
+		  "password": {
+			"type": "string",
+			"description": "The integration user password"
```

Comment:
[ARMMustGuidelineViolation] Plese note that secrets are not meant to be returned with a GET operation it must be exposed with a POST operation. 

this is explained here : 

[https://eng.ms/docs/products/arm/rp_onboarding/process/property_design_best_practices](https://eng.ms/docs/products/arm/rp_onboarding/process/property_design_best_practices)

Secrets
Secrets should never be exposed on a GET. The preferred pattern for cases where a secret is required on write (e.g. adminPasssword) is to return a null value on GET for the resource. If a secret does need to be returned after the fact, a POST api can be used to allow for granular RBAC. If this needs to be returned in a template, a list* (e.g. listKeys) name exposes it to the template language. Properties that contain a secret whether on PUT, GET or POST, MUST be annotated with x-ms-secret: true in your Swagger. Examples of x-ms-secret usage.